Support `JavaVersion` in `setJavaVersion`

MattSturgeon opened the pull request:
As discussed in [this comment](https://github.com/firstdarkdev/modpublisher/issues/4#issuecomment-2002562432)

- [x] update javadoc
- [x] update README
- [x] ~~add unit test~~ update integration test

Closes #4

MattSturgeon committed:
[FEAT]: Support `JavaVersion` in `setJavaVersion`

Maintain support for "curseforge" format Java versions, e.g. "Java 8".

MattSturgeon commented on src/main/java/com/hypherionmc/modpublisher/plugin/ModPublisherGradleExtension.java:
```diff
@@ -226,12 +228,19 @@ public void setLoaders(ModLoader... loader) {
+    public void setJavaVersions(Object... version) {
+        for (Object v : version) {
+            // Handle strings where the java version is after a space, e.g. "Java 8"
+            if (v instanceof String) v = StringUtils.substringAfterLast((String) v, " ");
```
Would it be better to strip the `"Java "` prefix, rather than take the substring after last `" "`?

hypherionmc replied:
I think so yeah. That might be a "safer" option

MattSturgeon replied:
Done

MattSturgeon commented:
I've updated the testprojects, but I can't find any [TestKit](https://docs.gradle.org/current/userguide/test_kit.html) tests that use them...

I've updated the example in the README, let me know if you'd prefer a different example.

hypherionmc reviewed (approved):
These look good to me :)
